lang_qc/endpoints/pacbio_well.py

```diff
@@ -202,38 +192,29 @@ def get_seq_metrics(
     status_code=status.HTTP_201_CREATED,
 )
 def claim_qc(
-    run_name: str,
-    well_label: str,
+    id_product: str,
     user: User = Depends(check_user),
     qcdb_session: Session = Depends(get_qc_db),
     mlwhdb_session: Session = Depends(get_mlwh_db),
 ) -> QcState:
 
-    if (
-        WellWh(session=mlwhdb_session).well_exists(
-            run_name=run_name, well_label=well_label
-        )
-        is False
-    ):
-        raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND,
-            detail=f"Well {well_label} run {run_name} does not exist",
-        )
+    _validate_product_id_or_error(id_product)
+    mlwh_well = _find_well_product_or_error(id_product, mlwhdb_session)
 
-    well_qc = WellQc(session=qcdb_session, run_name=run_name, well_label=well_label)
+    well_qc = WellQc(session=qcdb_session, id_product=id_product, mlwh_well=mlwh_well)
     if well_qc.current_qc_state():
         raise HTTPException(
             status_code=status.HTTP_409_CONFLICT,
-            detail=f"Well {well_label} run {run_name} has already been claimed",
+            detail=f"Well for product {id_product} has already been claimed",
         )
 
     # Using default attributes for almost all arguments.
     # The new QC state will be set as preliminary.
-    return QcState.from_orm(well_qc.assign_qc_state(user=user))
+    return QcState.from_orm(well_qc.assign_qc_state(mlwh_well=mlwh_well, user=user))
```

Comment:
`mlwh_well` seems to be an argument for `well_qc.assign...`, and also an attribute in the construction of `well_qc`. Are these not redundant?

Reply:
Cannot see this as the object's property. What line is it?

Reply:
Ah, found in the code that creates the object